tox.ini

```diff
@@ -35,6 +35,14 @@ deps = -r{toxinidir}/requirements.txt
 commands =
   twine check dist/*
 
+[testenv:mypy]
+basepython = python3
+deps = -r{toxinidir}/requirements.txt
+       -r{toxinidir}/test-requirements.txt
+       mypy
+commands =
+  mypy {posargs} {toxinidir}/gitlab/base.py {toxinidir}/gitlab/__init__.py
```

Comment:
Should this wait for https://github.com/python-gitlab/python-gitlab/pull/1301 (same above)?

Reply:
Sure. Let's wait for that. Thanks!

Reply:
Merged so this can go ahead now if you have plans to add typing :)

Reply:
Thank you. I probably won't get time this weekend 😟

Reply:
No rush :)

Reply:
I just had a thought that since mypy runs really fast it could also be added to pre-commit hooks with https://github.com/pre-commit/mirrors-mypy. But this can be added later. 

It would just be added here to the existing config:
https://github.com/python-gitlab/python-gitlab/blob/5cc60d5a8ac129652611d3dc12b350b5ca7262b9/.pre-commit-config.yaml#L4-L22

Although not sure how many people really install the hooks since I still see a lot of failing lint checks with new PRs :thinking:  Maybe that needs to be more prominent.

Reply:
Agree that it could be added in the future. Not sure it should be added to this PR.